src/app/properties/property-detail/property-detail.component.ts

```diff
@@ -106,4 +108,19 @@ export class PropertyDetailComponent {
 
     this.price = sum;
   }
+
+  calculateAverageRating() {
+    if (!this.property || !this.property.reviews || this.property.reviews.length <= 0) {
+      return;
+    }
+
+    let sum = 0;
+    let count = 0;
+    this.property.reviews.forEach(review => {
+      sum += review.rating;
+      count++;
+    });
+
+    this.averageRating = sum / count;
+  }
```

Comment:
The average rating will probably be coming from the server, but it's OK to calculate like this for now.